### PR TITLE
add getter for response object

### PR DESCRIPTION
**Pull request by dready**
so people may introduce their own assertions

### CODE REVIEW

**Comment by BruceWouaigne** [comment]
Can somebody merge this one please :)

**Comment by klaussilveira** [comment]
I'm also interested in this merge. I don't see a reason for the response attribute to be private, also. It should be protected, so child contexts can work with it.

**Comment by dready** [comment]
hey guys.. come on it's been months and we are only talking about a getter.. i mean, it's just one click away...

what's the hold up?

**Comment by stevergill** [comment]
This would be really useful for me as well.

**Comment by dantleech** [comment]
This is pretty esential for writing custom assertions I think .. //cc @stof

File: src/Context/WebApiContext.php [hunk 1]
```diff
@@ -59,6 +59,14 @@ public function setClient(Client $client)
     }
 
     /**
+     * @return \GuzzleHttp\Message\ResponseInterface
+     */
+    public function getResponse()
```
Comment (dantleech) [inline src/Context/WebApiContext.php]: Should this be protected?

Comment (dready) [inline src/Context/WebApiContext.php]: why? everything else is public and i see no harm in that.

bests

**Comment by tomaszhanc** [comment]
Is there any reason why you don't wanna merged this one?

**Comment by klaussilveira** [comment]
Any news on this, @stof?

File: src/Context/WebApiContext.php (hunk 1 above)
Comment (dantleech) [inline src/Context/WebApiContext.php]: As far as I know it is not required outside of the scope of this class, in which case it is always better to encapsulate functionality.

**Comment by AndyWendt** [comment]
Please merge :smile:

File: src/Context/WebApiContext.php (hunk 1 above)
Comment (AndyWendt) [inline src/Context/WebApiContext.php]: Protected probably would make sense

**Comment by mathop** [comment]
It would be very useful for me too. Is there any reason to not merge it?

**Comment by AndyWendt** [comment]
The fact that it is public may be an issue since public visibility is probably not needed

**Comment by dbaltas** [comment]
+1 with `protected`

**Comment by ferodss** [comment]
Please merge :smile: [2]

**Comment by RTodorov** [comment]
Jun 2014 and it's not merged? Really?

:+1: here.......

**Comment by klaussilveira** [comment]
This PR will make anniversary soon. :birthday:

**Comment by AlmogBaku** [comment]
+1 for that!

File: src/Context/WebApiContext.php (hunk 1 above)
Comment (henriquemoody) [inline src/Context/WebApiContext.php]: Maybe you could do the same of #27 here:

```php
protected function getResponse()
{
    if (! $this->response) {
        throw new Exception("You must first make a request to check a response.");
    }

    return $this->response;
}
```

I don't understand why this was not even reviewed, it has 23 comments and almost a year here.
This feature is very useful for custom contexts... Any way, I will probably copy WebApiContext and use it as my base context until this merge... :cry:

**Comment by katalisha** [comment]
As a temporary solution you can always fork this project and fix it, then add the fork to your composer.json

    "repositories": [{
        "type": "vcs",
        "url": "https://github.com/xxxxxx"
    }],

Simply remove the repository when this gets merged/closed to go back to the source.

**Comment by farconada** [comment]
merge please

**Comment by farconada** [comment]
merge please

**Comment by AndyWendt** [comment]
:cake: Been over a year.  Happy birthday.

**Comment by yokomizor** [comment]
:cake:

**Comment by ferodss** [comment]
Happy birthday :cake:

**Comment by dbaltas** [comment]
Dear @stof , I consider it professional courtesy to at least reply to this thread.

**Comment by mathop** [comment]
:cake: :tada: :balloon:

**Comment by dready** [comment]
thx :cake:

**Comment by AlmogBaku** [comment]
IMO this PR should not approve, and we need to change this vars/methods to protected

**Comment by AndyWendt** [comment]
Changing the vars to protected would be ideal but there hasn't been ANY guidance given from the package maintainer.

**Comment by klaussilveira** [comment]
Happy birthday! :cake:

**Comment by xurumelous** [comment]
Hey guys, we are using this fork with Guzzle 6 support and protected client: https://github.com/LinioIT/behat-web-api-extension :)

**Comment by kipelovets** [comment]
:cake: 

I believe getResponse() should be public so that assertions could be added by other Context classes without inheritance.

Why isn't this PR merged yet?

**Comment by mattjanssen** [comment]
:+1: 

Please make the response object available to child classes. Protected or getter; doesn't matter either way. I've had to copy/paste the entire `WebApiContext` into my own code. Yuck! :speak_no_evil:

**Comment by mattrlong** [comment]
Heres a +1 from 2016. Approaching a two year old PR!

**Comment by AlmogBaku** [comment]
All these email notification just reminding me why I ditched PHP

**Comment by klaussilveira** [comment]
Should we organize a party on Jun 22?

**Comment by jkirkby91** [comment]
@AlmogBaku what because one library is being long about merging a pull request? if you want it that bad just fork it

**Comment by dready** [comment]
Since nobody cares i close this. It's been too long for just i minor merge. It really shows how to ignore the community. Sorry, but imho this really sucked.

**Pull request closed without merging**